tests/unit/wallet/test_transaction.py

```diff
@@ -532,3 +533,106 @@ async def test_basic_use_cases_sqlite(self):
+        self.assertListEqual(self.outputs(tx), [19.92])
+
+        await self.ledger.release_outputs(utxos)
+
+        # everything: one output with initial amount (0.0) bumped
+        tx = await self.tx(
+            [],  # inputs
+            [self.txo(0.0)],  # outputs
+            everything=True
+        )
+        self.assertListEqual(self.inputs(tx), avail)
+        self.assertListEqual(self.outputs(tx), [19.92])
+
+        await self.ledger.release_outputs(utxos)
+
+        # everything: two outputs with initial amounts bumped
+        tx = await self.tx(
+            [],  # inputs
+            [self.txo(1.0), self.txo(4.0)],  # outputs
+            everything=True
+        )
+        self.assertListEqual(self.inputs(tx), avail)
+        self.assertListEqual(self.outputs(tx), [8.46, 11.46])
+
+        await self.ledger.release_outputs(utxos)
+
+        # everything: some inputs provided
+        if self.ledger.coin_selection_strategy == 'sqlite':
+            # NOTE: With this strategy, get_spendable_utxos() grabs extra
+            # utxos with the plan that any excess change can be added to
+            # the outputs. Hence the given initial output must be less
+            # than the absolute maximum (19.92).
+            await self.ledger.reserve_outputs(utxos[2:]);
+            tx = await self.tx(
+                [self.txi(self.txo(a)) for a in avail[2:]],  # inputs
+                [self.txo(19.0)],  # outputs
+                everything=True
+            )
+            self.assertListEqual(self.inputs(tx), avail[2:] + avail[0:2])
+            # NOTE: The maximum amount (19.92) was transferred. But it
+            # was broken into two outputs (19.0 and 0.92).
+            self.assertListEqual(self.outputs(tx), [19.0, 0.92])
+        else:
+            await self.ledger.reserve_outputs(utxos[2:]);
+            tx = await self.tx(
+                [self.txi(self.txo(a)) for a in avail[2:]],  # inputs
+                [self.txo(19.92)],  # outputs
+                everything=True
+            )
+            self.assertListEqual(self.inputs(tx), avail[2:] + avail[0:2])
+            self.assertListEqual(self.outputs(tx), [19.92])
+
+        await self.ledger.release_outputs(utxos)
+
+        # everything: maximum output already present
+        tx = await self.tx(
+            [],  # inputs
+            [self.txo(19.92)],  # outputs
+            everything=True
+        )
+        self.assertListEqual(self.inputs(tx), avail)
+        self.assertListEqual(self.outputs(tx), [19.92])
+
+        await self.ledger.release_outputs(utxos)
+
+        # everything: insufficient funds
+        try:
+            tx = await self.tx(
+                [],            # inputs
+                [self.txo(19.93)],  # outputs
+                everything=True
+            )
+        except lbry.error.InsufficientFundsError:
+            pass
+        else:
+            self.fail("expected InsufficientFunds exception")
```

Comment:
Should this be an [assertRaises](https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertRaises)?

Reply:
Done.